### ISSUE TITLE
Configuration Error - Ubuntu

### ISSUE THREAD

**Issue by jamie-moriarty** [CMU-Perceptual-Computing-Lab/openpose#863]
### Issue Summary

Make is consistently failing at 2%. I have tried a fresh download of openpose, downloaded caffe from the caffe github, and tried the openpose caffe download.

### Executed Command (if any)
make -j

### Errors (if any)

jamie@linda-the-laptop:~/openpose/build$ make -j
Scanning dependencies of target openpose_caffe
[  1%] Creating directories for 'openpose_caffe'
[  1%] No download step for 'openpose_caffe'
[  2%] No update step for 'openpose_caffe'
[  2%] No patch step for 'openpose_caffe'
[  2%] Performing configure step for 'openpose_caffe'
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at /usr/share/cmake-3.10/Modules/FindBoost.cmake:567 (message):
  Imported targets and dependency information not available for Boost version
  (all versions older than 1.33)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindBoost.cmake:907 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake-3.10/Modules/FindBoost.cmake:1558 (_Boost_MISSING_DEPENDENCIES)
  cmake/Dependencies.cmake:8 (find_package)
  CMakeLists.txt:46 (include)

CMake Warning at /usr/share/cmake-3.10/Modules/FindBoost.cmake:567 (message):
  Imported targets and dependency information not available for Boost version
  (all versions older than 1.33)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindBoost.cmake:907 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake-3.10/Modules/FindBoost.cmake:1558 (_Boost_MISSING_DEPENDENCIES)
  cmake/Dependencies.cmake:8 (find_package)
  CMakeLists.txt:46 (include)

CMake Warning at /usr/share/cmake-3.10/Modules/FindBoost.cmake:567 (message):
  Imported targets and dependency information not available for Boost version
  (all versions older than 1.33)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindBoost.cmake:907 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake-3.10/Modules/FindBoost.cmake:1558 (_Boost_MISSING_DEPENDENCIES)
  cmake/Dependencies.cmake:8 (find_package)
  CMakeLists.txt:46 (include)

-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
CMake Error at /usr/share/cmake-3.10/Modules/FindBoost.cmake:1947 (message):
  Unable to find the requested Boost libraries.

  Unable to find the Boost header files.  Please set BOOST_ROOT to the root
  directory containing Boost or BOOST_INCLUDEDIR to the directory containing
  Boost's headers.
Call Stack (most recent call first):
  cmake/Dependencies.cmake:8 (find_package)
  CMakeLists.txt:46 (include)

-- Found GFlags: /usr/include  
-- Found gflags  (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libgflags.so)
-- Found Glog: /usr/include  
-- Found glog    (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libglog.so)
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.10/Modules/FindProtobuf.cmake:543 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/ProtoBuf.cmake:4 (find_package)
  cmake/Dependencies.cmake:43 (include)
  CMakeLists.txt:46 (include)

-- Configuring incomplete, errors occurred!
See also "/home/jamie/openpose/build/caffe/src/openpose_caffe-build/CMakeFiles/CMakeOutput.log".
See also "/home/jamie/openpose/build/caffe/src/openpose_caffe-build/CMakeFiles/CMakeError.log".
CMakeFiles/openpose_caffe.dir/build.make:105: recipe for target 'caffe/src/openpose_caffe-stamp/openpose_caffe-configure' failed
make[2]: *** [caffe/src/openpose_caffe-stamp/openpose_caffe-configure] Error 1
CMakeFiles/Makefile2:99: recipe for target 'CMakeFiles/openpose_caffe.dir/all' failed
make[1]: *** [CMakeFiles/openpose_caffe.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

### Type of Issue
- Compilation/installation error

### Your System Configuration
1. **Whole console output** : from openpose/build/caffe/src/openpose-caffe-build/CMakeFiles/CMakeOutput.log 
[FULL ERROR OUTPUT HERE](https://pastebin.com/mR9UQ9G4)

2. **OpenPose version**: Latest GitHub

3. **General configuration**:
    - **Installation mode**: CMake
    - **Operating system** : Ubuntu 18.04.1 LTS
    - **Release or Debug mode**? : Release
    - Compiler : gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0

4. **Non-default settings**:
    - **3-D Reconstruction module added**? (by default: no):  Yes

5. **3rd-party software**:
    - **Caffe version**: Default from OpenPose
    - **CMake version** : cmake version 3.10.2
    - **OpenCV version**: pre-compiled `apt-get install libopencv-dev`

7. If **CPU-only mode** issue:
    - **CPU brand & model**: product: Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz
    - Total **RAM memory** available: ~3.8 GiB of 7.7 GiB

**Comment by gineshidalgo99** [comment]
Hi, I believe Ubuntu 18 is not compatible with Caffe. The errors are coming from there.

I'd recommend to try to install your own Caffe. If you are able, then simply link OpenPose CMake-gui with that Caffe.

**Comment by jamie-moriarty** [comment]
I was unable to get caffe working on 18, after reinstalling with 16.04 I was able to get the install functioning. I did run into some issues where I had installed the librealsense sdk 2.0 before installing caffe and openpose which caused some issues because they use different versions of protobuf. Installing caffe and openpose before the librealsense sdk fixed this. Just in case anyone else runs into that problem.
